app/common/network/network-model.ts

```diff
@@ -0,0 +1,366 @@
+import { createEffect, createEvent, createStore, sample, scopeBind } from 'effector';
+import { keyBy } from 'lodash-es';
+import { combineEvents, spread } from 'patronum';
+
+import { type ApiPromise } from '@polkadot/api';
+
+import { DEFAULT_CHAINS } from '@/common/utils/chains';
+import { chainsService, metadataService } from '@/services';
+import { providerService } from '@/services/network/provider-service';
+import { type Chain, type ChainMetadata } from '@/types/substrate';
+
+import { networkUtils } from './network-utils';
+import { type Connection, ConnectionStatus, type ProviderWithMetadata } from './types';
+
+const networkStarted = createEvent<'chains_dev' | 'chains_prod'>();
+const chainConnected = createEvent<ChainId>();
+const chainDisconnected = createEvent<ChainId>();
+const connectionStatusChanged = createEvent<{ chainId: ChainId; status: ConnectionStatus }>();
+
+const connected = createEvent<ChainId>();
+const disconnected = createEvent<ChainId>();
+const failed = createEvent<ChainId>();
+
+const $chains = createStore<Record<ChainId, Chain>>({});
+const $connections = createStore<Record<ChainId, Connection>>({});
+
+const $metadata = createStore<ChainMetadata[]>([]);
+const $metadataSubscriptions = createStore<Record<ChainId, VoidFunction>>({});
+
+const populateChainsFx = createEffect(async (file: 'chains_dev' | 'chains_prod'): Promise<Record<ChainId, Chain>> => {
+  const chains = await chainsService.getChainsData({ file, sort: true });
+
+  return keyBy(chains, 'chainId');
+});
+
+const getConnectedChainIndicesFx = createEffect((): Promise<ChainIndex[]> => {
+  const cloud = '3_0,2,3;19_0,2,3;'; // Karura & unknown chain
+  const chainIndices = cloud.match(/(\d+)_/g)?.map(match => match[0]);
+
+  return Promise.resolve(chainIndices || []);
```

Comment:
Maybe add comments here to clarify why we are doing it this way?

Reply:
I've added down below 😌 `// TODO: uncomment during task - https://app.clickup.com/t/869502m30`
Right now it's just a placeholder imitating logic with CloudStorage.